core/psh/nc.c

```diff
@@ -0,0 +1,337 @@
+/*
+ * Phoenix-RTOS
+ *
+ * nc - TCP and UDP connect or listen
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <sys/select.h>
+
+#include "psh.h"
+
+
+/*
+ * This could be achieved using getaddrinfo(),
+ * however it does not recognize scope id for now
+ * and requires dynamic memory, which we
+ * are trying to avoid.
+ */
+static int psh_nc_sockaddrFill(int *af_family, char *addrstr, char *portstr,
+                               struct sockaddr_storage *sockaddr,
+                               socklen_t *addrlen)
+{
+	struct sockaddr_in6 *saddr_in6;
+	struct sockaddr_in *saddr;
+	char *ifname, *endptr;
+	int port = 0, scopeid = 0;
+	uint32_t addr[4] = { };
+
+	if (*af_family == AF_UNSPEC) {
+		if (addrstr && strchr(addrstr, ':') != NULL)
```

Comment:
Maybe it would be better to explicity compare the pointer with NULL.
```
		if ((addrstr != NULL) && (strchr(addrstr, ':') != NULL))
````
This also applies to other pointer comparisions.

Reply:
For me, it's fine either way. I can change it and compare it to NULL.

Reply:
I don't know if this is required but pointers are compared to `NULL`.

Reply:
Yes, pointers should be always compared to NULL. It's not really about correctness but rather about readability.

Reply:
Although, for most of the cases, I prefer being explicit, in this case this comparison looks more readable for me. I guess, it's because of the time I spent writing in Python. ;) But, if it is prefered in our code, I'll fix this to compare to NULL and get used to it.